packages/svelte/src/internal/client/dom/operations.js

```diff
@@ -78,7 +78,9 @@ export function child(node) {
 export function first_child(fragment, is_text) {
 	if (!hydrating) {
 		// when not hydrating, `fragment` is a `DocumentFragment` (the result of calling `open_frag`)
-		return /** @type {DocumentFragment} */ (fragment).firstChild;
+		var first = /** @type {DocumentFragment} */ (fragment).firstChild;
+		if (first instanceof Comment) return first.nextSibling;
```

Comment:
Is this guaranteed to work? What if someone uses the `preserveComments` option and starts the child content with a comment?

Reply:
good point. we actually already have a version of this bug with hydration comments IIRC — one option would be to ban comments whose data is the empty string or `"]"` or whatever. Another would be to add an argument to `first_child`

Reply:
added a condition that `first.data === ''`, will leave the other stuff as a TODO for later since it concerns hydration anyway